### PR TITLE
Improve stateless test runner

### PR DESCRIPTION
**Pull request by alexkats**
### Changelog category (leave one):
- CI Fix or Improvement (changelog entry is not required)

<details><summary>CI</summary>

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [x]  Allow: Stateless tests
- [x]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All Builds
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Run only fuzzers related jobs (libFuzzer fuzzers, AST fuzzers, etc.)
- [ ]  Exclude: AST fuzzers
---
- [ ]  Do not test
- [x]  Woolen Wolfdog
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

</details>

### CODE REVIEW

**Comment by robot-ch-test-poll** [comment]
*This is an automated comment for commit e5bd09b6c389588a1d3feb487e9fb23288e30771 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/69864/e5bd09b6c389588a1d3feb487e9fb23288e30771/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Stateless tests</td><td>Runs stateless functional tests for ClickHouse binaries built in various configurations -- release, debug, with sanitizers, etc</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/69864/e5bd09b6c389588a1d3feb487e9fb23288e30771/stateless_tests__release__old_analyzer__s3__databasereplicated__[2_2].html">✅ success</a></td></tr>
<tbody>
</table>
</details>

**Comment by alexkats** [comment]
Stateless tests (tsan) [1/4] - https://github.com/ClickHouse/ClickHouse/issues/49156

File: tests/docker_scripts/process_functional_tests_result.py [hunk 1]
```diff
@@ -161,6 +164,24 @@ def process_result(result_path, broken_tests):
             retries,
             test_results,
         ) = process_test_log(result_path, broken_tests)
+
+        # Check test_results.tsv for sanitizer asserts, crashes and other critical errors
+        if test_results_path and os.path.exists(test_results_path):
+            with open(test_results_path, "r", encoding="utf-8") as test_results_file:
```
Comment (tavplubix) [inline tests/docker_scripts/process_functional_tests_result.py:170]: Do we already have `test_results.tsv` at this point? AFAIK, `https://github.com/ClickHouse/ClickHouse/blob/9b55180184c0f38cf88bf5902687603cf903ee40/tests/docker_scripts/process_functional_tests_result.py#L222` parses run.log in order to generate and write `test_results.tsv`:
https://github.com/ClickHouse/ClickHouse/blob/9b55180184c0f38cf88bf5902687603cf903ee40/tests/docker_scripts/process_functional_tests_result.py#L222

Comment (alexkats) [reply tests/docker_scripts/process_functional_tests_result.py:170]: I reordered lines in `stateless_runner.sh`, and `test_results.tsv` is generated by `check_logs_for_critical_errors` from `stress_tests.lib`, then it is used in `process_functional_tests_result.py` and the whole `test_results.tsv` is regenerated

Comment (tavplubix) [reply tests/docker_scripts/process_functional_tests_result.py:170]: Ah, ok, this was confusing, please add a comment. Does it make sense to add `--in-results-file` to make it more explicit?

Comment (alexkats) [reply tests/docker_scripts/process_functional_tests_result.py]: Elaborated a bit on it in the comment, added the argument as well

**Review by tavplubix — APPROVED** [review]

**Pull request merged**